explainers/anti-tracking.md

```diff
@@ -0,0 +1,247 @@
+# Anti-tracking in signed packages
+
+<!-- TOC depthTo:3 -->
+
+- [Blocking credentials](#blocking-credentials)
+  - [Possible spellings](#possible-spellings)
+  - [Tentatively-rejected spellings](#tentatively-rejected-spellings)
+- [Blocking normal link decoration](#blocking-normal-link-decoration)
+- [Blocking privileged link decoration](#blocking-privileged-link-decoration)
+- [Privileged server-side collaboration](#privileged-server-side-collaboration)
+- [Appendices](#appendices)
+  - [Security concerns](#security-concerns)
+  - [Behavior on an unexpected response](#behavior-on-an-unexpected-response)
+
+<!-- /TOC -->
+
+This explainer attempts to solve the problems identified in issues
+[#422](https://github.com/WICG/webpackage/issues/422) and
+[#423](https://github.com/WICG/webpackage/issues/423). Specifically, we're
+worried that a distributor might be able to use the fact that it can choose
+which exchange or package, signed by a publisher, to serve in response to a
+client's request, to transfer its notion of the user's identity to the
+publisher.
+
+The approach this document takes is to prevent the distributor from accessing
+its notion of the user's identity. There are several routes by which a
+distributor could learn the user's identity:
+
+1. The browser could send the distributor its top-level credentials. [Countermeasures](#blocking-credentials)
+1. The source of a link could use the same link-decoration tools as it might use
+   in a cross-origin link, which other efforts are blocking. Note that the
+   source of a link and the distributor of a package could be the same origin.
+   [Countermeasures](#blocking-normal-link-decoration)
+1. The source of a link could decorate links to its distributor to send a user
+   ID in ways an unrelated cross-origin target couldn't be expected to receive.
+   [Countermeasures](#blocking-privileged-link-decoration)
+1. The source of a link could collaborate with its distributor on the server
+   side to send a user ID in ways an unrelated cross-origin target couldn't be
+   induced to collaborate.
+   [Discussion](#privileged-server-side-collaboration)
+
+These are tackled individually in the following sections.
+
+## Blocking credentials
+
+The navigation to a signed package (or exchange) must be done without
+credentials. This is roughly [credentials
+mode](https://fetch.spec.whatwg.org/#concept-request-credentials-mode) ==
+`"omit"`, with changes to Fetch to make it work for navigations.
```

Comment:
Ideally we'd be able to talk about which network partition this uses too, but I'm not sure if the spec framework for that exists yet. :-/

Reply:
Which do we want it to use? I think it's the uncredentialed partition from the origin of the SXG, and the contents get to cross into the credentialed partition for the origin that signed them?

Reply:
There isn't really a singular uncredentialed partition right now. We do partition credentialed and uncredentialed connections, but it was more a consequence of not confusing connection-based auth (client certs, NTLM, Channel ID at one point, etc) than an actual partition. In theory a smarter connection pool could pool across those without messing that up.

I don't think anyone is partitioning the HTTP cache by credentialed vs uncredentialed. (I'm *sure* sites always correctly labels their `Vary: Cookie` resources. :-) ) And pages can freely make uncredentialed subresource requests and either write data via URLs or read data via CORS-readable responses. I believe that means credentialed vs uncredentialed is useful for threat models that consider intentional state mechanisms (so CSRF and such), but that it's not useful for threat models which consider random caches being misused as state (which we seem to be converging on for anti-tracking).

https://github.com/kinu/speculative-loading talks about using a temporary partition, which would probably require a bit of spec machinery if that's what we settle on.

Reply:
I've added some handwaving about figuring this out. I'll merge this now, but we can update it as we get the spec text worked out.